server/api-util/lineItemHelpers.js

```diff
@@ -0,0 +1,192 @@
+  const numericTotalPrice = new Decimal(numericPrice)
+    .times(percentage)
+    .dividedBy(100)
+    .toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+/**
+ * Calculates lineTotal for lineItem based on seats and units.
+ * The total will be `unitPrice * units * seats`.
+ *
+ * @param {Money} unitPrice
+ * @param {int} unitCount
+ * @param {int} seats
+ *
+ * @returns {Money} lineTotal
+ */
+exports.calculateTotalPriceFromSeats = (unitPrice, unitCount, seats) => {
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const numericTotalPrice = new Decimal(numericPrice)
+    .times(unitCount)
+    .times(seats)
+    .toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+/**
+ * Calculates the quantity based on the booking start and end dates depending on booking type.
+ *
+ * @param {Date} startDate
+ * @param {Date} endDate
+ * @param {string} type
+ *
+ * @returns {number} quantity
+ */
+exports.calculateQuantityFromDates = (startDate, endDate, type) => {
+  if (type === LINE_ITEM_NIGHT) {
+    return nightsBetween(startDate, endDate);
+  } else if (type === LINE_ITEM_DAY) {
+    return daysBetween(startDate, endDate);
+  }
+  throw new Error(`Can't calculate quantity from dates to unit type: ${type}`);
+};
+
+/**
+ *
+ *  `lineTotal` is calculated by the following rules:
+ * - If `quantity` is provided, the line total will be `unitPrice * quantity`.
+ * - If `percentage` is provided, the line total will be `unitPrice * (percentage / 100)`.
+ * - If `seats` and `units` are provided the line item will contain `quantity` as a product of `seats` and `units` and the line total will be `unitPrice * units * seats`.
+ *
+ * @param {Object} lineItem
+ * @return {Money} lineTotal
+ *
+ */
+exports.calculateLineTotal = lineItem => {
+  const { code, unitPrice, quantity, percentage, seats, units } = lineItem;
+
+  if (quantity) {
+    return this.calculateTotalPriceFromQuantity(unitPrice, quantity);
+  } else if (percentage) {
+    return this.calculateTotalPriceFromPercentage(unitPrice, percentage);
+  } else if (seats && units) {
+    return this.calculateTotalPriceFromSeats(unitPrice, units, seats);
+  } else {
+    throw new Error(
+      `Can't calculate the lineTotal of lineItem: ${code}. Make sure the lineItem has quantity, percentage or both seats and units`
+    );
+  }
+};
+
+/**
+ * Calculates the total sum of lineTotals for given lineItems
+ *
+ * @param {Array} lineItems
+ * @retuns {Money} total sum
+ */
+exports.calculateTotalFromLineItems = lineItems => {
+  const numericTotalPrice = lineItems.reduce((sum, lineItem) => {
+    const lineTotal = this.calculateLineTotal(lineItem);
+    const numericPrice = convertMoneyToNumber(lineTotal);
+    return new Decimal(numericPrice).add(sum);
+  }, 0);
+
+  const unitPrice = lineItems[0].unitPrice;
```

Comment:
I think there should be a check that we are checking correct line-item (I guess "line-item/night") and not trust that first line-item is always pointing to that instead of being a "line-item/provider-commission" or something.

Reply:
D'oh, never mind. this was just about checking currency!